### PR TITLE
feat(header-animations): L3-3037 Header Animations Transitions

### PR DESCRIPTION
**Pull request by adietrich3074**
**Jira ticket**

[L3-3037](https://phillipsauctions.atlassian.net/browse/L3-3037)

**Summary**

Added css transitions and animations to desktop and mobile header.

**Change List (describe the changes made to the files)**

- Slide-up and slide-down animations added to accordions on mobile menu
- Slide out from left animation added to main menu on mobile
- Mobile search slide out animation and input styling fixed on mobile
- Slide-up and slide-down animations added on hover to Auctions and Departments nav items

**Acceptance Test (how to verify the PR)**

- Verify that the animations match expected behavior from [desktop protopie](https://cloud.protopie.io/p/1b0a7def5ee11e505e4808b4) and [mobile protopie](https://cloud.protopie.io/p/322a8627be4733482918eebf?sceneId=be3f9870-99e5-41f8-aaee-7e17394dba93)

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-3037]: https://phillipsauctions.atlassian.net/browse/L3-3037?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by scottdickerson** [comment]
The font weight of the search placeholder and text looks off: 
image

**Comment by scottdickerson** [comment]
I'm seeing some weirdness with the mobile menu extending past the storybook:
image

File: src/components/Navigation/NavigationItemTrigger/NavigationItemTrigger.tsx [hunk 1]
```diff
@@ -45,15 +47,15 @@ const NavigationItemTrigger = forwardRef<HTMLLIElement, NavigationItemTriggerPro
...
-        </SSRMediaQuery.Media>
+        {/* <SSRMediaQuery.Media lessThan="md"> */}
+        <MobileNavigationItemTrigger id={id} label={label} {...commonProps}>
+          {navListElement
+            ? React.cloneElement(navListElement[0], {
+                className: `${baseClassName}__submenu--mobile`,
+              })
+            : undefined}
+        </MobileNavigationItemTrigger>
+        {/* </SSRMediaQuery.Media> */}
```
Comment (scottdickerson) [inline src/components/Navigation/NavigationItemTrigger/NavigationItemTrigger.tsx:58]: did this need to be removed?  Ideally it wouldn't render to the DOM for accessibility/screen readers

**Review by scottdickerson — CHANGES_REQUESTED** [review]
some UX changes

**Comment by adietrich3074** [comment]
> The font weight of the search placeholder and text looks off: image

I think this is caused by the scaling of the font token I was asked to change it to but this is correct based on the values for that input field in the figma:
`color: var(--Interactive-Static-CTA, #000);

/* Text/String 1 (Bold) */
font-family: var(--Font-Family-Body, Montserrat);
font-size: var(--Font-Size-Text-1, 1rem);
font-style: normal;
font-weight: 600;
line-height: var(--Font-Line-Height-String-1, 1.25rem); /* 125% */
letter-spacing: var(--Font-Character-Spacing-Text, 0rem);`

**Comment by adietrich3074** [comment]
> I'm seeing some weirdness with the right side of the mobile menu extending past the right edge of the screen: image

This looks like an issue with the desktop container in storybook. I'm not sure how to fix this because the issue doesn't occur in the regular dev tools view of the header or on my phone. My phone even displays the component in storybook without this misalignment in both portrait and landscape view

Screenshot_20240924_104108_Chrome
Screenshot_20240924_103743_Chrome

The above screenshots are from my phone

**Comment by scottdickerson** [comment]
> > The font weight of the search placeholder and text looks off: image
> 
> I think this is caused by the scaling of the font token I was asked to change it to but this is correct based on the values for that input field in the figma: `color: var(--Interactive-Static-CTA, #000);
> 
> /* Text/String 1 (Bold) _/ font-family: var(--Font-Family-Body, Montserrat); font-size: var(--Font-Size-Text-1, 1rem); font-style: normal; font-weight: 600; line-height: var(--Font-Line-Height-String-1, 1.25rem); /_ 125% */ letter-spacing: var(--Font-Character-Spacing-Text, 0rem);`

@adietrich3074 can you reach out to design/UX to double check this to see if it's a token issue or this is a snowflake that we need to override

**Review by scottdickerson — APPROVED** [review]
LGTM

**Pull request merged**